lcls_tools/common/devices/magnet.py

```diff
@@ -267,9 +273,32 @@ def scan(
         function: Optional[callable] = None,
     ) -> None:
         for setting in scan_settings:
-            self.bctrl = setting
+            try:
+                self.set_bdes_for_scan(setting)
+            except MagnetTimeoutError as e:
+                # TODO: use logging here instead
+                print("Error:", e)
             function() if function else None
 
+    def set_bdes_for_scan(self, bval: float, settle_timeout_in_seconds: int = 5):
+        self.bdes = bval
+        self.trim()
+        time_when_trim_started = datetime.now()
+        while not self.is_bact_settled():
```

Comment:
Is this code repeated around ~350 for a set of magnets? Maybe we should pull it out as a function?